aws/scenarios/vm/vm/vm.go

```diff
@@ -0,0 +1,51 @@
+package vm
+
+import (
+	"github.com/DataDog/test-infra-definitions/aws"
+	awsEc2 "github.com/DataDog/test-infra-definitions/aws/ec2/ec2"
+	"github.com/DataDog/test-infra-definitions/command"
+	"github.com/DataDog/test-infra-definitions/common/config"
+
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type VM struct {
+	Context     *pulumi.Context
+	Runner      *command.Runner
+	Environment *config.CommonEnvironment
+	// TODO add file manager as soon as https://github.com/DataDog/test-infra-definitions/pull/9 is merged
+	// FileManager   *command.FileManager
+	DockerManager *command.DockerManager
+}
+
+/*
+NewVM Creates an instance of VM on aws
+Returns a VM object with most operation mamagers to install docker containers
+and run generic commands.
+*/
+func NewVM(ctx *pulumi.Context) (*VM, error) {
```

Comment:
I think this fits very well with the `VM` interface and provider idea.
We delegate the creation of the `VM` to the called method (like `NewEC2Instance`), and it will return a VM with already instantiated `Runner` and `PackageManager`.

Reply:
@vboulineau : Can you elaborate on  `provider idea.`?

Reply:
Well, a `VM` provider would be basically anything that returns a `VM` interface. It could be a function or a struct it does not really matter.
If we need to we could standardize through a `VMProvider` interface, add registration and a single main to choose a VM provider from `datadog-agent/CLI` but it may not be necessary just yet.

Reply:
Addressed and discussed at https://github.com/DataDog/test-infra-definitions/pull/10#discussion_r1037404760